src/main/java/net/apnic/whowas/domain/controller/DomainRouteController.java

```diff
@@ -0,0 +1,60 @@
+package net.apnic.whowas.domain.controller;
+
+import javax.servlet.http.HttpServletRequest;
+
+import net.apnic.whowas.history.ObjectClass;
+import net.apnic.whowas.history.ObjectKey;
+import net.apnic.whowas.rdap.controller.RDAPControllerUtil;
+import net.apnic.whowas.rdap.TopLevelObject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Rest Controller for the RDAP /domain path segment.
+ *
+ * Controller is responsible for dealing with current state RDAP path segments.
+ */
+@RestController
+@RequestMapping("/domain")
+public class DomainRouteController
+{
+    private final static Logger LOGGER = LoggerFactory.getLogger(DomainRouteController.class);
+
+    private RDAPControllerUtil rdapControllerUtil = null;
+
+    @Autowired
+    public DomainRouteController(RDAPControllerUtil rdapControllerUtil)
+    {
+        this.rdapControllerUtil = rdapControllerUtil;
+    }
+
+    @RequestMapping(value="/{handle:.+}", method=RequestMethod.GET)
+    public ResponseEntity<TopLevelObject> domainPathGet(
+        HttpServletRequest request,
+        @PathVariable("handle") String handle)
+    {
+        LOGGER.info("domain GET path query for {}", handle);
```

Comment:
Info logging is probably too high here. I'd recommend debug level. (copy-paste comment for the other initial LOGGER.info's)